### PR TITLE
Add paging to //schematics.

### PR DESCRIPTION
**Pull request by kenzierocks**
Re-implementation of #294. ~~Untested.~~ Working perfectly.

### CODE REVIEW

File: worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java [hunk 1]
```diff
@@ -244,14 +251,16 @@ public void formats(Actor actor) throws WorldEditException {
...
+            flags = "dnp",
             help = "List all schematics in the schematics directory\n" +
                     " -d sorts by date, oldest first\n" +
-                    " -n sorts by date, newest first\n"
+                    " -n sorts by date, newest first\n" +
+                    " -p <page> prints the requested page\n"
     )
     @CommandPermissions("worldedit.schematic.list")
-    public void list(Actor actor, CommandContext args) throws WorldEditException {
+    public void list(Actor actor, CommandContext args, @Switch('p') @Optional("1") int page) throws WorldEditException {
```
Comment (TomyLobo) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java]: hmm does that @Optional("1") mean "1 optional argument to -p" or does it mean "-p is optional and requires one argument".
In the first case, your code doesn't match your help blurb
In the second case, @Optional is veeeery strange...

Comment (kenzierocks) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java]: 3) `@Optional("1")` means the default value is `"1"`.

Comment (TomyLobo) [inline worldedit-core/src/main/java/com/sk89q/worldedit/command/SchematicCommands.java]: oh, for some reason I missed those quotes. makes sense, don't mind me :)

**Comment by kenzierocks** [comment]
@wizjany or @sk89q does this look good to merge?

**Pull request merged**